src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateBuilder.java

```diff
@@ -166,7 +168,7 @@ public Pod build() {
             }
         }
 
-        volumes.put(WORKSPACE_VOLUME_NAME, template.getWorkspaceVolume().buildVolume(WORKSPACE_VOLUME_NAME, slave != null ? slave.getPodName() : null));
+        volumes.put(WORKSPACE_VOLUME_NAME, template.getWorkspaceVolume().buildVolume(WORKSPACE_VOLUME_NAME, agent != null ? agent.getPodName() : null));
```

Comment:
```suggestion
        volumes.put(WORKSPACE_VOLUME_NAME, template.getWorkspaceVolume().buildVolume(WORKSPACE_VOLUME_NAME, agent.getPodName()));
```

if it is really `@Nonnull`?

Reply:
The old constructor is used in other places so I have to keep it `@CheckForNull`